python/ray/serialization.py

```diff
@@ -202,12 +215,22 @@ def _deserialize_object(self, data, metadata, object_id):
                 try:
                     in_band, buffers = unpack_pickle5_buffers(data)
                     if len(buffers) > 0:
-                        return pickle.loads(in_band, buffers=buffers)
+                        obj = pickle.loads(in_band, buffers=buffers)
                     else:
-                        return pickle.loads(in_band)
+                        obj = pickle.loads(in_band)
                 # cloudpickle does not provide error types
                 except pickle.pickle.PicklingError:
                     raise DeserializationError()
+
+                if object_id.is_nil():
+                    assert len(self.get_and_clear_contained_object_ids()) == 0
```

Comment:
When could this happen?

Reply:
This happens when arguments are inlined. Added a comment.